File: src/main/java/my/little/changelog/validator/VersionValidator.java
package my.little.changelog.validator;

import com.google.common.collect.Lists;
import com.google.inject.Singleton;
import lombok.extern.log4j.Log4j2;
import my.little.changelog.annotation.NotNull;
import my.little.changelog.error.CustomError;
import my.little.changelog.error.ErrorMessage;
import my.little.changelog.model.project.Version;

import java.util.List;

@Singleton
@Log4j2
public class VersionValidator {

    /**
     * Validate model.
     * @param version version model.
     * @return error messages.
     */
    @NotNull
    public List<CustomError> validate(Version version) {
        return Lists.newArrayList();
    }
}

File: src/main/kotlin/my/little/changelog/validator/VersionValidator.kt
package my.little.changelog.validator

import my.little.changelog.model.auth.User
import my.little.changelog.model.version.Version
import my.little.changelog.persistence.repo.VersionRepo

object VersionValidator {

    fun validateLatest(version: Version, user: User): ValidatorResponse {
        val errors = mutableListOf<String>()
        if (version.id != VersionRepo.findLatestByProject(version.project).id) {
            errors.add("Version is not latest!")
        }
        return ValidatorResponse(errors)
    }
}
